src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -84,7 +84,10 @@ class MinikubeImpl implements Minikube {
 }
 
 async function minikubeUpgradeAvailable(context: Context): Promise<void> {
-    if (!await checkPresent(context, CheckPresentMode.Silent)) {
+
+    const checkForUpgrade = vscode.workspace.getConfiguration().get("vsminikube.checkForUpgrade");
+
+    if (!await checkPresent(context, CheckPresentMode.Silent) || !checkForUpgrade) {
```

Comment:
Either amend the comment on the next line, or break this out into two statements:

```
if (!await ...) {
  // not installed...
  return;
}
if (!checkForUpgrade) {
  // don't bother the user
  return;
}
```

Perhaps even better introduce an explaining variable:

```
const performUpgradeCheck = await checkPresent(...) && checkForUpgrade;
if (!performUpgradeCheck) {
  return;
}
```